subsys/usb/device_next/class/usbd_hid.c

```diff
@@ -235,7 +235,7 @@ static int handle_get_report(const struct device *dev,
 	const uint8_t id = HID_GET_REPORT_ID(setup->wValue);
 	struct hid_device_data *const ddata = dev->data;
 	const struct hid_device_ops *ops = ddata->ops;
-	const size_t size = net_buf_tailroom(buf);
+	const size_t size = setup->wLength;
```

Comment:
Not sure if using `setup->wLength` directly is appropriate, a potential alternative would be `MIN(setup->wLength, net_buf_tailroom(buf))`. Host can submit any wLength value up to 65535 regardless the actual buffer size. Currently the stack will fail if it cannot allocate such large request even if the actual payloads are much smaller.

Reply:
I can use `MIN(setup->wLength, net_buf_tailroom(buf)` here. Still, I wonder if the condition should not be validated somewhere in lower layers (so that USB class implementation could already assume that `setup->wLength` is proper.

Reply:
This is more of a wondering how this issue should be solved in general. This sort of mechanism currently results in USB2CV MSC failure with default build settings because the testcase is requesting string descriptors length with 4096 wLength whereas the strings are significantly shorter than that. While the strings would definitely fit inside the much smaller buffer, they are not even given a chance because default UDC buf pool is failing to allocate 4096 bytes long buffer.